src/WebComponent.js

```diff
@@ -188,14 +195,18 @@ export class WebComponent extends HTMLElement {
     if (typeof this.template === "string") {
       this.innerHTML = this.template;
     } else if (typeof this.template === "object") {
+      let host = this;
+      if (this.shadowMode) {
+        host = this.attachShadow({ mode: this.shadowMode });
+      }
       const tree = this.template;
 
       // TODO: smart diffing
       if (JSON.stringify(this.#prevDOM) !== JSON.stringify(tree)) {
         const el = createElement(tree);
         if (el) {
-          if (Array.isArray(el)) this.replaceChildren(...el);
-          else this.replaceChildren(el);
+          if (Array.isArray(el)) host.replaceChildren(...el);
```

Comment:
I think array is no longer expected since we now return a document fragment for siblings with no parent?